modules/integration/tests-common/integration-test-utils/src/main/java/org/wso2/appserver/integration/common/utils/CarbonCommandToolsUtil.java

```diff
@@ -0,0 +1,280 @@
+        long time = System.currentTimeMillis() + DEFAULT_START_STOP_WAIT_MS;
+        log.info("Shutting down server..");
+        boolean logOutSuccess = false;
+        if (ClientConnectionUtil.isPortOpen(Integer.parseInt(ExtensionConstants.
+                                                                     SERVER_DEFAULT_HTTPS_PORT))) {
+
+            int httpsPort = Integer.parseInt(FrameworkConstants.SERVER_DEFAULT_HTTPS_PORT) + portOffset;
+            String url = automationContext.getContextUrls().getBackEndUrl();
+            String backendURL = url.replaceAll("(:\\d+)", ":" + httpsPort);
+
+            ServerAdminClient serverAdminServiceClient = new ServerAdminClient(backendURL,
+                              automationContext.getContextTenant().getTenantAdmin().getUserName(),
+                              automationContext.getContextTenant().getTenantAdmin().getPassword());
+
+            serverAdminServiceClient.shutdown();
+
+            while (System.currentTimeMillis() < time && !logOutSuccess) {
+                logOutSuccess = isServerDown(automationContext, portOffset);
+                // wait until server shutdown is completed
+            }
+            log.info("Server stopped successfully...");
+        }
+
+    }
+
+    /**
+     * This method is to merge two arrays together
+     *
+     * @param parameters - Server startup arguments
+     * @param cmdArray   - Server startup command
+     * @return - merged array
+     */
+    private static String[] mergePropertiesToCommandArray(String[] parameters, String[] cmdArray) {
+        if (parameters != null && cmdArray != null) {
+            cmdArray = ArrayUtils.addAll(cmdArray, parameters);
+        }
+        return cmdArray;
+    }
+
+    /**
+     * This method is to check whether server is down or not
+     *
+     * @param automationContext - AutomationContext
+     * @param portOffset        - port offset
+     * @return boolean - if server is down true : else false
+     * @throws XPathExpressionException
+     */
+    public static boolean isServerDown(AutomationContext automationContext, int portOffset)
+            throws XPathExpressionException, InterruptedException {
+        boolean isPortOpen = true;
+        long startTime = System.currentTimeMillis();
+        // Looping the isPortOpen method, waiting for a while  to check the server is down or not
+        while (isPortOpen && (System.currentTimeMillis() - startTime) < TIMEOUT) {
+            isPortOpen = ClientConnectionUtil.isPortOpen(
+                    Integer.parseInt(FrameworkConstants.SERVER_DEFAULT_HTTPS_PORT) + portOffset);
+            if(isPortOpen){
+                Thread.sleep(1000); // waiting 1 sec to check isPortOpen again
+            }
+        }
+        return !isPortOpen;
+    }
+
+    /**
+     * This method is to return operating system
+     *
+     * @return if current os is windows return true : else false
```

Comment:
wrong method comment - not returning any boolean values. but method description indicate it returns true or false